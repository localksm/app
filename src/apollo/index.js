import {
  ApolloClient,
  ApolloLink,
  HttpLink,
  InMemoryCache,
} from 'apollo-boost';
import { WebSocketLink } from 'apollo-link-ws';
import { getMainDefinition } from 'apollo-utilities';
import { onError } from 'apollo-link-error';
import { setContext } from 'apollo-link-context';
import { split } from 'apollo-link';
import ls from 'react-native-local-storage';
import {
  GRAPHQL_ENDPOINT,
  GRAPHQL_SUBSCRIPTIONS_ENDPOINT,
} from '../utils/config';
import { ContextProvider, ContextConsumer, withContext } from './context';
import ApolloState from './stateManager';
import {
  SIGNUP,
  LOGIN,
  ADD_FUNDS,
  CONFIRM_PROPOSAL,
  SEND_ACCEPTANCE,
  SEND_RESOLUTION,
  SEND_SETTLEMENT,
  SEND_FULFILLMENT,
  INSERT_PROPOSAL_PAYMENT_METHOD,
  SEND_DISBURSEMENT_BUYER,
  SEND_DISBURSEMENT_SELLER,
  SELL,
} from './mutations';
import {
  VERIFY_USER,
  FEE,
  PAYMENT_METHODS,
  CURRENCIES,
  COUNTRIES,
  PUBLIC_KEY,
  QUERY_PROPOSALS,
  QUERY_USER_PROPOSALS,
} from './queries';

const httpLink = new HttpLink({
  uri: `${GRAPHQL_ENDPOINT}`,
});

const wsLink = new WebSocketLink({
  uri: `${GRAPHQL_SUBSCRIPTIONS_ENDPOINT}`,
  options: { reconnect: true },
});

export const setSession = async payload => {
  return await ls.save('session', payload);
};

export const getSession = async () => {
  return await ls.get('session');
};

export const removeSession = async () => {
  return await ls.clear();
};

const authLink = setContext(async _ => {
  const data = await getSession();
  const { session } = data !== null && data;
  const jwt = session && session.token;

  return {
    headers: {
      authorization: `Bearer ${jwt}`,
    },
  };
});

const httpAuthLink = authLink.concat(httpLink);

const errorLink = onError(({ graphQLErrors, networkError }) => {
  if (graphQLErrors) {
    graphQLErrors.map(({ message, locations, path }) =>
      // eslint-disable-next-line no-console
      console.log(
        `[GraphQL error]: Message: ${message}, Location: ${JSON.stringify(locations)}, Path: ${path}`,
      ),
    );
  }
  // eslint-disable-next-line no-console
  if (networkError) console.log(`[Network error]: ${networkError}`);
});

const link = split(
  ({ query }) => {
    const definition = getMainDefinition(query);
    return (
      definition.kind === 'OperationDefinition' &&
      definition.operation === 'subscription'
    );
  },
  wsLink,
  httpAuthLink,
);
export const cache = new InMemoryCache();
export const client = new ApolloClient({
  link: ApolloLink.from([errorLink, link]),
  cache,
  resolvers: {
    logout: async id => {
      await removeSession();
    },
  },
});

export const QUERIES = {
  VERIFY_USER,
  FEE,
  PAYMENT_METHODS,
  COUNTRIES,
  CURRENCIES,
  PUBLIC_KEY,
  QUERY_PROPOSALS,
  QUERY_USER_PROPOSALS,
};

export const MUTATIONS = {
  SIGNUP,
  LOGIN,
  SEND_ACCEPTANCE,
  SEND_RESOLUTION,
  SEND_SETTLEMENT,
  SEND_FULFILLMENT,
  INSERT_PROPOSAL_PAYMENT_METHOD,
  ADD_FUNDS,
  CONFIRM_PROPOSAL,
  SEND_DISBURSEMENT_BUYER,
  SEND_DISBURSEMENT_SELLER,
  SELL,
};

export const SUBSCRIPTIONS = {};

export { ContextConsumer, ContextProvider, withContext };

const initialState = {
  session: null,
  offer: null,
  filterData: [],
  navigation: null,
  evidenceImages: null,
  token: null,
};
cache.writeData({ data: initialState });

export const state = ApolloState;
