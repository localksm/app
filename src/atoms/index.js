export { default as Button } from './Button';
export { default as InputText } from './InputText';
export { default as Link } from './Link';
export { default as Dropdown } from './Dropdown';
export { default as FBLoginButton } from './facebookButton';
export { default as GoogleButton } from './GoogleButton';
export { default as ButtonSignUp } from './ButtonSignUp';
export { default as Fees } from './Fees';
export { default as ButtonSignIn } from './ButtonSignIn';
export { default as DropdownPaymentMethods } from './DropdownPaymentMethods';
export { default as DropdownCountries } from './DropdownCountries';
export { default as DropdownCurrencies } from './DropdownCurrencies';
export { default as AddFundsButton } from './AddFundsButton';
export { default as QrWallet } from './QrCode';
export { default as ProposalTypeImages } from './ProposalTypeImages'; 
export { default as ConfirmSentBuyButton } from './ConfirmSentBuyButton';
export { default as ConfirmReceivedButton } from './ConfirmReceivedButton';
export { default as TWLoginButton } from './TwitterButton';
export { default as SellButton } from './SellButton';
export { default as InputLayout} from './InputLayout';