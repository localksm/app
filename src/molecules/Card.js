import React from 'react';
import { TouchableOpacity, StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import moment from 'moment';
import Offer from './Offer';
import { mapPaymentMethod, cardProposalNavigation } from '../utils/misc';

function CardComponent({ item }) {
  const navigation = useNavigation();

  return (
    <TouchableOpacity
      testID="press-card"
      onPress={() => {
        cardProposalNavigation(item, navigation);
      }}>
      <Offer
        payment={mapPaymentMethod(item.body?.paymentMethod)}
        usernameMaker={item.body?.usernameMaker}
        date={moment(new Date(item.body?.updatedAt).toUTCString()).format(
          'MMMM Do YYYY - h:mm:ss A',
        )}
        offered={item.body?.requestAmount}
        requiered={item.body?.offerAmount}
        status={item.status}
        currency={
          item.body?.operationType === 'add_funds' ||
          item.body?.operationType === 'buy'
            ? item.body?.offerAsset
            : item.body?.requestAsset
        }
        isOffer={true}
        operationType={item.body?.operationType}
      />
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  listFooterComponent: {
    height: 30,
  },
  contentContainer: {
    paddingBottom: 300,
  },
});

export default CardComponent;
