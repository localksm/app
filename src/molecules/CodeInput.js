import React, { useState, useEffect } from 'react';
import { Text, View, TextInput, StyleSheet } from 'react-native';

const CodeInput = props =>{
    const [codeArr, setCodeArr] = useState([]);
    const [currentIndex, setCurrentIndex] = useState(0);
    let codeInputRefs = [];

    useEffect(() => {
        setCodeArr(new Array(props.codeLength).fill(''));        
    }, []);

    const _clear =()=> {    
        setCodeArr(new Array(props.codeLength).fill(''));
        setCurrentIndex(0);
        _setFocus(0);
    }

    const _setFocus =(index)=> {        
        codeInputRefs[index].focus();
    }
      
    const _blur=(index) => {
        codeInputRefs[index].blur();
    }

    const _onFocus =(index) =>{
        const { onCodeChange } = props;
        codeInputRefs[index].clear();
        let newCodeArr = codeArr;        
        const currentEmptyIndex = codeArr.findIndex( element => element =='');
        if (currentEmptyIndex !== -1 && currentEmptyIndex < index) {
          return _setFocus(currentEmptyIndex);
        }
        for (let i in newCodeArr) {
          if (i >= index) {
            newCodeArr[i] = '';
          }
        }

        setCodeArr(newCodeArr);
        setCurrentIndex(index);
        onCodeChange(newCodeArr.join(''))
      }

    const _onInputCode =(character, index) => {
        const { codeLength, onCodeChange } = props;
        let newCodeArr = codeArr;
        newCodeArr[index] = character;
        if (index === codeLength - 1) {            
            _blur(currentIndex);            
          } else {
            _setFocus(currentIndex + 1);
          }

        setCodeArr(newCodeArr);
        setCurrentIndex( currentIndex + 1);
        onCodeChange(newCodeArr.join(''))

    }

    const _getClassStyle =(active, id) => {
        const isNotEmpty = codeArr.reduce((accumulator, currentValue, index) =>{
            if(currentValue !== "") 
              accumulator.push(index);
            return accumulator; 
          },new Array()).includes(id);

        return Object.assign({},styles.activeDotStyle,{ 
            backgroundColor: isNotEmpty?'white': 'transparent',
            borderWidth: active? 1 : 0.2,
         });
    }

    let codeInputs = [];
    for (let index = 0; index < props.codeLength; index++) {
        const id = index;
        codeInputs.push(
            <TextInput 
                key={id}
                keyboardType="numeric"
                ref={ref => (codeInputRefs[id] = ref)}
                style={_getClassStyle(currentIndex == id, id )}
                onChangeText={text => _onInputCode(text, id)}
                autoFocus={false}
                onFocus={() => _onFocus(id)}
                selectTextOnFocus={false}
                clearTextOnFocus={true}
                selectionColor={'white'}
                underlineColorAndroid="transparent"
                maxLength={1} />
        );
    }
   return (
        <View style={styles.container}>
            {codeInputs}
        </View>
    );
}

const styles = StyleSheet.create({
    activeDotStyle:{
        borderWidth: 0.2,
        borderRadius: 20,
        marginLeft: 20,
        height: 20,
        width: 20,
        borderColor:'white',
        fontSize: 5
    },
    container:{
        flexDirection:'row',
        marginVertical:10,
        
    }
});

export default CodeInput;
