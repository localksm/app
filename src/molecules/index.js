export { default as HeaderLayout } from './HeaderLayout';
export { default as Offer } from './Offer';
export { default as FooterWhite } from './FooterWhite';
export { default as MultiView } from './MultiView';
export { default as CompletedMultiView } from './CompletedMultiView';
export { default as FilePickerComponent } from './FilePickerComponent';
export { default as FilePicker } from './FilePicker';
export { default as WalletContent } from './WalletContent';
export { default as CardProposal } from './CardProposal';
export { default as Card } from './Card';
export { default as BalanceHeader } from './BalanceHeader';
export { default as PaymentForm } from './PaymentForm';
export { default as PaymentFormInputs } from './PaymentFormInputs';
export { default as FormInput } from './FromInput';
export { default as FormTextInp } from './FormTextInp';
export { default as FormTextAreaInp } from './FormTextAreaInp';
export { default as GenericGQLLoading } from './GenericGQLLoading';
export { default as GenericGQLError } from './GenericGQLError';
export { default as ConfirmedBuySellButton } from './ConfirmedBuySellButton';
export { default as WithdrawText } from './WithdrawText';
export { default as WalletSwitch} from './WalletSwitch';
export { default as WalletViewButton } from './WalletViewButton';