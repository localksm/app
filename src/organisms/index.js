export { default as HomeLayout } from './HomeLayout';
export { default as OfferList } from './OfferList';
export { default as OfferDetails } from './OfferDetails';
export { default as OfferDetailContent } from './OfferDetailContent';
export { default as FormSignIn } from './FormSignIn';
export { default as FormSignInContent } from './FormSignInContent';
export { default as FormSignUp } from './FormSignUp';
export { default as FormCreateOfferBuy } from './FormCreateOfferBuy';
export { default as FormCreateOfferSell } from './FormCreateOfferSell';
export { default as ConfirmationBuySell } from './ConfirmationBuySell';
export { default as ConfirmedBuySell } from './ConfirmedBuySell';
export { default as AcceptedBuySell } from './AcceptedBuySell';
export { default as DisbursementBuyer } from './DisbursementBuyer';
export { default as DisbursementSeller } from './DisbursementSeller';
export { default as Completed } from './Completed';
export { default as WalletView } from './WalletView';
export { default as Report } from './Report';
export { default as ReportContent } from './ReportContent';
export { default as MediationView } from './MediationView';
export { default as MediationResult } from './MediationResult';
export { default as MyOffer } from './MyOffer';
export { default as Slider } from './Slider';
export { default as OnboardingImage } from './OnboardingImage';
export { default as HelpView } from './HelpView';
export { default as FormLayout } from './FormLayout';
export { default as Withdraw } from './Withdraw';
export { default as FormCreatePin } from './FormCreatePin';
export { default as EnterPin } from './EnterPin';
export { default as SendSettlementMakerContent } from './SendSettlementMakerContent';

export { default as CustomDrawerContent } from './CustomDrawer';
