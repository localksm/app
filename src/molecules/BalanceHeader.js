import React from 'react';
import { View, StyleSheet, Text, ActivityIndicator } from 'react-native';
import Balance from '../atoms/Balance';
import { getSession } from '../apollo';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    paddingBottom: '10%',
  },
  textBalance: {
    color: 'white',
    fontSize: 18,
    fontFamily: 'Poppins-Medium',
  },
  offerList: {
    borderTopRightRadius: 50,
    borderTopLeftRadius: 50,
    justifyContent: 'center',
    paddingTop: '1%',
    paddingBottom: '80%',
    backgroundColor: 'white',
  },
});

function BalanceHeader() {
  const [session, setSession] = React.useState(null);
  const [show, showBalance] = React.useState(false);

  React.useEffect(() => {
    set();
  }, []);

  async function set() {
    const { session } = await getSession();
    setSession(session);
  }

  return session !== null ? (
    <View style={styles.container}>
      <Text style={{ ...styles.textBalance}}>
        Balance
      </Text>
      <Balance
        id={session.id}
        styleTotal={{ ...styles.textBalance, fontSize: 20 }}
        styleFree={{ ...styles.textBalance, fontSize: 10 }}
        styleText={{ ...styles.textBalance, fontSize: 14 }}
      />
    </View>
  ) : (
    <ActivityIndicator />
  );
}

export default BalanceHeader;
