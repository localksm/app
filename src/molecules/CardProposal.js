import React, { useEffect, useState } from 'react';
import {
  Text,
  View,
  FlatList,
  TouchableOpacity,
  StyleSheet,
  ActivityIndicator,
  Alert,
} from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { useQuery } from '@apollo/react-hooks';
import moment from 'moment';
import { QUERIES, getSession } from '../apollo';
import Offer from './Offer';

function Loading() {
  return (
    <View style={[styles.container, { marginVertical: 30 }]}>
      <ActivityIndicator size="large" color="black" />
    </View>
  );
}

function Error({ error }) {
  return (
    <View style={styles.container}>
      <Text>{`Error! ${error.message}`}</Text>
    </View>
  );
}

function mapPaymentMethod(method) {
  const methods = {
    VE: 'Venmo',
    ZE: 'Zelle',
    MP: 'Mercado Pago',
    WU: 'Western Union',
    MG: 'Money Gram',
    NE: 'Neteller',
    UP: 'Uphold',
    PP: 'Paypal',
    BN: 'Bank',
    OT: 'Other',
  };
  return methods[method];
}

const CardProposal = props => {
  const [id, setId] = useState(null);
  const navigation = useNavigation();
  
  useEffect(() => {
    set();
  }, []);

  async function set() {
    const { session } = await getSession();
    setId(session.id);
  }

  const { loading, error, data } = useQuery(QUERIES.QUERY_PROPOSALS, {
    variables: {userId: id},
    pollInterval: 3000
  });
  if (loading) return <Loading />;
  if (error) return <Error error={error} />;

  return (
    <>
      <FlatList
        data={data.proposals}
        ListFooterComponent={<View />}
        ListFooterComponentStyle={styles.listFooterComponent}
        contentContainerStyle={styles.contentContainer }
        renderItem={({ item }) => {
          return (
            <TouchableOpacity
              onPress={() => {
                switch (item.body.operationType) {
                  case 'add_funds':
                    return navigation.navigate('DetailsOffer', { ...item });
                  case 'withdraw_funds':
                    return navigation.navigate('DetailsOffer', { ...item });
                  case 'buy':
                    return navigation.navigate('DetailsOffer', { ...item });
                  case 'sell':
                    return navigation.navigate('DetailsOffer', { ...item });
                  default:
                    Alert.alert(
                      'Warning!',
                      'Unexpected error, contact the support area',
                    );
                    break;
                }
              }}>
              <Offer
                payment={mapPaymentMethod(item.body.paymentMethod)}
                usernameMaker={item.body.usernameMaker}
                date={moment(new Date(item.body.updatedAt).toUTCString()).format(
                  'MMMM Do YYYY - h:mm:ss A',
                )}
                offered={item.body.requestAmount}
                requiered={item.body.offerAmount}
                status={item.status}
                currency={item.body.offerAsset}
                isOffer={true}
                operationType={item.body.operationType}
              />
            </TouchableOpacity>
          );
        }}
      />
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  },
  listFooterComponent:{
    height: 30
  },
  contentContainer:{
    paddingBottom: 300 
  }
});

export default CardProposal;
