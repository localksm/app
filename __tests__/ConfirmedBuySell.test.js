import 'react-native';
import React from 'react';
import { act, create } from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import { fireEvent, render } from 'react-native-testing-library';

import Adapter from 'enzyme-adapter-react-16';

import { ConfirmedBuySell } from '../src/organisms';
import { ConfirmedBuySellButton } from '../src/molecules';

configure({ adapter: new Adapter() });

const mockParams = {
  params: {
    status: 'accepted',
    body: {
      usernameMaker: 'maker',
      offerAsset: 'native',
      offerAmount: 0,
      paymentMethod: 'VE',
      operationType: 'buy',
      paymentData: {
        accountNumber: 'accountNumber',
        address: 'address',
        bankData: '12345',
        email: 'email',
        lastName: 'lastName',
        name: 'name',
        phone: '',
        proposalId: 1,
      },
    },
  },
};

it('confirmation buy renders correctly', async () => {
  jest.useFakeTimers();
  await act(() => {
    create(<ConfirmedBuySell route={mockParams} />);
  });
});

it('confirmation sell renders correctly', async () => {
  jest.useFakeTimers();
  await act(() => {
    mockParams.params.body.operationType = 'sell';
    mockParams.params.status = 'confirmed';
    create(<ConfirmedBuySell route={mockParams} />);
  });
});

it('test report problem on press', async () => {
  jest.useFakeTimers();
  const openHandler = jest.fn();

  const { getByTestId } = render(
    <ConfirmedBuySell navigation={{ navigate: () => {} }} route={mockParams} />,
  );

  fireEvent.press(getByTestId('link'));
  expect(openHandler).not.toHaveBeenCalled();
});

it('test confirmation alert', async () => {
  jest.useFakeTimers();
  await act(() => {
    const wrapper = shallow(<ConfirmedBuySellButton send={true} />);

    expect(wrapper).toMatchSnapshot();
  });
});

it('confirmation buy button renders correctly with buy and accepted', async () => {
  jest.useFakeTimers();
  await act(() => {
    create(
      <ConfirmedBuySellButton
        send={false}
        status="accepted"
        operationType="buy"
        variables={{}}
        navigation={{ navigate: () => {} }}
        params={mockParams}
      />,
    );
  });
});

it('confirmation buy button renders correctly with buy and confirmed', async () => {
  jest.useFakeTimers();
  await act(() => {
    create(
      <ConfirmedBuySellButton
        send={false}
        status="confirmed"
        operationType="buy"
        variables={{}}
        navigation={{ navigate: () => {} }}
        params={mockParams}
      />,
    );
  });
});