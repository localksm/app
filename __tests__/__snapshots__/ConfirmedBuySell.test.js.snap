// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test confirmation alert 1`] = `ShallowWrapper {}`;
